bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTHandler.java

```diff
@@ -0,0 +1,267 @@
+ */
+@NonNullByDefault
+public class SAICiSMARTHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SAICiSMARTHandler.class);
+
+    @Nullable
+    SAICiSMARTVehicleConfiguration config;
+    private @Nullable Future<?> pollingJob;
+    private ZonedDateTime lastAlarmMessage = ZonedDateTime.ofInstant(Instant.EPOCH, ZoneId.systemDefault());
+
+    // if the binding is initialized, treat the car as active to get some first data
+    private ZonedDateTime lastCarActivity = ZonedDateTime.now();
+
+    public SAICiSMARTHandler(HttpClientFactory httpClientFactory, Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(SAICiSMARTBindingConstants.CHANNEL_FORCE_REFRESH) && command == OnOffType.ON) {
+            // reset channel to off
+            updateState(CHANNEL_FORCE_REFRESH, OnOffType.from(false));
+            // update internal activity date, to query the car for about a minute
+            notifyCarActivity(ZonedDateTime.now().minus(9, ChronoUnit.MINUTES), true);
+        } else if (channelUID.getId().equals(CHANNEL_SWITCH_AC) && command == OnOffType.ON) {
+            // reset channel to off
+            updateState(CHANNEL_SWITCH_AC, OnOffType.ON);
+            new Thread(() -> {
+                // enable air conditioning
+                try {
+                    sendACCommand((byte) 5, (byte) 8);
+                } catch (URISyntaxException | ExecutionException | TimeoutException | InterruptedException e) {
+                    logger.error("A/C On Command failed", e);
+                }
+            }).start();
+        } else if (channelUID.getId().equals(CHANNEL_SWITCH_AC) && command == OnOffType.OFF) {
+            // reset channel to off
+            updateState(CHANNEL_SWITCH_AC, OnOffType.OFF);
+            new Thread(() -> {
+                // disable air conditioning
+                try {
+                    sendACCommand((byte) 0, (byte) 0);
+                } catch (URISyntaxException | ExecutionException | TimeoutException | InterruptedException e) {
+                    logger.error("A/C Off Command failed", e);
+                }
+            }).start();
+        } else if (channelUID.getId().equals(CHANNEL_LAST_ACTIVITY) && command instanceof DateTimeType) {
+            // update internal activity date from external date
+            notifyCarActivity(((DateTimeType) command).getZonedDateTime(), true);
+        }
+    }
+
+    protected @Nullable SAICiSMARTBridgeHandler getBridgeHandler() {
+        return (SAICiSMARTBridgeHandler) super.getBridge().getHandler();
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SAICiSMARTVehicleConfiguration.class);
+
+        ThingBuilder thingBuilder = editThing();
+        thingBuilder.withChannels(thing.getChannels().stream().filter(c -> {
+            // TODO: check properties
```

Comment:
Could this todo be fixed?

Reply:
I have removed the filter and kept the sun roof in the list of channels.  It will not work for some cars but not all channels will work on all cars... so this should be OK for the user.